spark3-extensions/src/main/scala/org/apache/iceberg/spark/extensions/IcebergSparkSessionExtensions.scala

```diff
@@ -37,6 +37,6 @@ class IcebergSparkSessionExtensions extends (SparkSessionExtensions => Unit) {
     // TODO: PullupCorrelatedPredicates should handle row-level operations
     extensions.injectOptimizerRule { _ => PullupCorrelatedPredicatesInRowLevelOperations }
     extensions.injectOptimizerRule { _ => RewriteDelete }
-    extensions.injectPlannerStrategy { _ => ExtendedDataSourceV2Strategy }
+    extensions.injectPlannerStrategy { spark => ExtendedDataSourceV2Strategy(spark) }
```

Comment:
Did the apply method not resolve here? Wondering why we needed to modify the anonymous arg

Reply:
This follows what we do above, and it seemed reasonably clean to me. I guess we could just pass `ExtendedDataSourceV2Strategy` and have apply automatically called.

Reply:
Ah I was just following the pattern with injectOptimizer lines directly above